### ISSUE TITLE
Bug - Installation Failure - SQL Error

### ISSUE THREAD

**Issue by sarshad01** [ballerine-io/ballerine#2721]
### Description
`pnpm run kyb-manual-review-example` results in failure, more specifically database connection error  when `@ballerine/workflows-service@0.7.37 db:reset` is invoked. Running  `npm db:reset` executes successfully if ran manually within services/workflows-service directory. However, running `npm run seed` throws Foreign key constraint error (Alert_businessId_fkey (index))

### Expected Behaviour
BackOffice, Collection and Workflows services should start listening

### Actual Behaviour
Command Fails with Exit code 1

### Environment
* Node version: v20.17.0
* Device, operating system and its version: Ubuntu 22.04.4 LTS
  *
* Browser and its version: Firefox 130.0
  *
* Version of the affected apps and packages: Dev Branch @ c39a8df
  *

### Steps to Reproduce
1. Follow the steps here : https://docs.ballerine.com/en/getting_started/installation/
2. `pnpm run kyb-manual-review-example` would exit the process with the database connection error
3. Run `npm run db:reset` while the db container is running and the command will succeed
4. Run `npm run seed` this will fail showing foreign key error

### Any possible solutions
1- My assumption was the database connection error occurs due to localhost being used, I switched it to 127.0.0.1 in the /services/workflows-service/.env however it did not solve the issue.

2- I then tried to run `npm run db:reset` manually and it worked

3- I then manually ran `npm run seed` which failed throwing foreign key constraint error

### Examples and references

Output of `pnpm run kyb-manual-review-example` :

![Screenshot from 2024-09-18 19-28-50](https://github.com/user-attachments/assets/36a2f3e4-a764-4212-bb76-3cbeb2e49af6)

 

Output of  `npm run seed` : 

![Screenshot from 2024-09-18 19-32-58](https://github.com/user-attachments/assets/d0564f21-3853-4806-bfa1-943a41f5c1c9)

**Comment by WFA-juliver** [comment]
is this first run?

**Comment by sarshad01** [comment]
@WFA-juliver  I have tried multiple times and unsuccessful before trying to install it via docker

**Comment by WFA-juliver** [comment]
@sarshad01 when running the example, did you verify if theres a db container with postgres running?

try

```
rm -rf node_modules \
&& pnpm install \
&& pnpm monorepo:init \
&& pnpm run kyb-manual-review-example

```
and check if theres database running `docker ps -a`

**Comment by sarshad01** [comment]
@WFA-juliver Yes I did and the db container is running. The workflows-service:setup basically starts the db container via docker:db command, so even if it is not running, it starts running when the workflows-service is setup. If it is already running, the command restarts it. My assumption is the 'wait' for whatever reason does not work and the script tries to run migrations before the container is ready for accepting connections?

**Comment by WFA-juliver** [comment]
> @WFA-juliver Yes I did and the db container is running. The workflows-service:setup basically starts the db container via docker:db command, so even if it is not running, it starts running when the workflows-service is setup. If it is already running, the command restarts it. My assumption is the 'wait' for whatever reason does not work and the script tries to run migrations before the container is ready for accepting connections?

yeah I see that too, hope someone on the ballerine team can help us. I really wanted to make this software up and running on my instance.

**Comment by WFA-juliver** [comment]
@sarshad01  any luck on your side?

**Comment by sarshad01** [comment]
I was able to get it up and running by manually running the db commands, to fix the `npm seed` and SQL error  I edited /services/workflows-service/scripts/seed.ts and commented the seedTransactionsAlerts block

**Comment by WFA-juliver** [comment]
@sarshad01 how about the docker deployment, you have luck?

**Comment by TestYouTestUntilTest** [comment]
Created a special sh script /home/XXXXXXXXX/ballerine/services/workflows-service/scripts/waitDb.sh

#!/bin/bash
```

set -e

# There are some times database is not ready yet!
# We'll check if database is ready and we can connect to it
# then the rest of the code run as well.

if [[ -z "${DB_HOST}" ]]; then
  DB_HOST="localhost"
  echo "set DB_HOST=localhost"
else
  DB_HOST="${DB_HOST}"
fi

if [[ -z "${DB_PORT}" ]]; then
  DB_PORT="5432"
  echo "set DB_PORT=5432"
else
  DB_PORT="${DB_PORT}"
fi

echo "Waiting for database..."
echo DB_NAME: ${DB_NAME}
echo DB_HOST: ${DB_HOST}
echo DB_PORT: ${DB_PORT}
while ! ncat -z ${DB_HOST} ${DB_PORT};
do
  sleep 1
  echo "."
done
echo "Connected to database."

echo "Extra wait for 5 second(s)"
sleep 5
echo "Go forward...."

# ... Run what you have to here
```


Then add a lauch of this script to /home/XXXXXXXXX/ballerine/services/workflows-service/scripts/db-reset.ts

```
  console.info('Before Running Prisma migrate reset check db server...');
  //https://nodejs.org/api/child_process.html#child_processexecsynccommand-options
  execSync('bash $HOME/ballerine/services/workflows-service/scripts/waitDb.sh', { stdio: 'inherit' });
```

Actually, 3-5 seconds delay is the solution

**Comment by WFA-juliver** [comment]
@TestYouTestUntilTest I'm stuck on this error

```
Migration `20230723212353_add_plv8` failed to apply cleanly to the shadow database.
Error:
db error: ERROR: permission denied to create extension "plv8"
HINT: Must be superuser to create this extension.
   0: schema_core::state::DevDiagnostic
             at schema-engine/core/src/state.rs:266

```

any idea?

**Comment by TestYouTestUntilTest** [comment]
> @TestYouTestUntilTest I'm stuck on this error
> 
> ```
> Migration `20230723212353_add_plv8` failed to apply cleanly to the shadow database.
> Error:
> db error: ERROR: permission denied to create extension "plv8"
> HINT: Must be superuser to create this extension.
>    0: schema_core::state::DevDiagnostic
>              at schema-engine/core/src/state.rs:266
> ```
> 
> any idea?

Add sudo to the shell script to run commands and enter your user pass during script running.
But actually you can reduce script just for waiting 5 seconds (and this will not require any sudo)

echo "Extra wait for 5 second(s)"
sleep 5
echo "Go forward...."

**Comment by changtraisitinh** [comment]
i got error when try run command `pnpm run kyc-manual-review-example`. please help


```
Applying migration `20230723212353_add_plv8`
Error: P3018
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
Migration name: 20230723212353_add_plv8
Database error code: 58P01
Database error:
ERROR: could not access file "$libdir/plv8-3.2.3": No such file or directory
DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E58P01), message: "could not access file \"$libdir/plv8-3.2.3\": No such file or directory", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("dfmgr.c"), line: Some(210), routine: Some("internal_load_library") }
```
